pkg/api/common/net.go

```diff
@@ -37,3 +41,17 @@ func IP4BroadcastAddress(n *net.IPNet) net.IP {
 	}
 	return last
 }
+
+// GetVNETSubnetIDComponents extract subscription, resourcegroup, vnetname, subnetname from the vnetSubnetID
+func GetVNETSubnetIDComponents(vnetSubnetID string) (string, string, string, string, error) {
+	vnetSubnetIDRegex := `^\/subscriptions\/([^\/]*)\/resourceGroups\/([^\/]*)\/providers\/Microsoft.Network\/virtualNetworks\/([^\/]*)\/subnets\/([^\/]*)$`
+	re, err := regexp.Compile(vnetSubnetIDRegex)
+	if err != nil {
+		return "", "", "", "", err
+	}
+	submatches := re.FindStringSubmatch(vnetSubnetID)
+	if len(submatches) != 4 {
+		return "", "", "", "", fmt.Errorf("unable to find 4 submatches")
```

Comment:
changed the return here because `return "", "", "", "", fmt.Errorf("unable to find 4 submatches") ` would return a nil error